src/DynamoCore/Interfaces/IWorkspaceModel.cs

```diff
@@ -7,6 +7,7 @@ namespace Dynamo.Interfaces
     public interface IWorkspaceModel
     {
         string Name { get; }
+        string FileName { get; set; }
```

Comment:
Okay, for now I'd like you to cast to `CustomNodeWorkspaceModel` in `CustomNodePathRemapper`.  This sucks, but it isolates the issue to some of our client code as opposed to the interface. I've set up tasks to repair this in the future.

Exposing the setter like this is a bad model, because the setter does not do any legality checks on the `FileName`. Furthermore, I'd like to move towards a model where we record mutations like this through the `RecordableCommand` framework.